src/support.js

```diff
@@ -246,14 +246,71 @@ Module['registerFunctions'] = registerFunctions;
 #endif // RELOCATABLE
 #endif // EMULATED_FUNCTION_POINTERS
 
+#if 0
+// TODO Currently logical operations like '#if a && b' or '#if a || b' are not
```

Comment:
yeah, this is an annoying current limitation. duplicating code might be the best option currently, but another one is to define new variables near the top of `src/jsifier.js` (around `HAS_MAIN`, which is similar). So you can define say `WASM_BACKEND_RESERVED_FUNCTION_POINTERS` to be true when wasm backend + there are reserved function pointers, and so forth. Or maybe `WASM_BACKEND_WITH_RESERVED_FUNCTION_POINTERS`, `WASM_BACKEND_WITHOUT_RESERVED_FUNCTION_POINTERS`?

I think it might be better that way, but not sure.